src/networking.c

```diff
@@ -3780,13 +3790,13 @@ void clientCommand(client *c) {
         listRewind(server.clients, &li);
         while ((ln = listNext(&li)) != NULL) {
             client *client = listNodeValue(ln);
-            if (addr && strcmp(getClientPeerId(client), addr) != 0) continue;
-            if (laddr && strcmp(getClientSockname(client), laddr) != 0) continue;
-            if (type != -1 && getClientType(client) != type) continue;
-            if (id != 0 && client->id != id) continue;
-            if (user && client->user != user) continue;
-            if (c == client && skipme) continue;
-            if (max_age != 0 && (long long)(commandTimeSnapshot() / 1000 - client->ctime) < max_age) continue;
+            if (client_filter.addr && strcmp(getClientPeerId(client), client_filter.addr) != 0) continue;
+            if (client_filter.laddr && strcmp(getClientSockname(client), client_filter.laddr) != 0) continue;
+            if (client_filter.type != -1 && getClientType(client) != client_filter.type) continue;
+            if (client_filter.id != 0 && client->id != client_filter.id) continue;
+            if (client_filter.user && client->user != client_filter.user) continue;
+            if (c == client && client_filter.skipme) continue;
+            if (client_filter.max_age != 0 && (long long)(commandTimeSnapshot() / 1000 - client->ctime) < client_filter.max_age) continue;
 
             /* Kill it. */
             if (c == client) {
```

Comment:
I mean, here, use the shared code for checking if it's a match, something like

```suggestion
            if (!clientMatchesFilter(c, client_filter) continue;

            /* Kill it. */
            if (c == client) {
```